### PR TITLE
Build everything with single dotnet build run, skip incremental checks for pack and test

### PR DESCRIPTION
**Pull request by pakrym**
Skips as many `dotnet *` invocations and incremental checks as possible

@pranavkm @victorhurdugaci

### CODE REVIEW

File: build/shade/_k-standard-goals.shade [hunk 1]
```diff
@@ -107,30 +109,42 @@ default PACKAGELIST_JSON_FILENAME = 'NuGetPackageVerifier.json'
         GitCommand("rev-parse HEAD >> " + commitHashFile);
         commitHash = File.ReadAllLines(commitHashFile)[0];
     }
-  
-    var projectFiles = Files.Include("src/*/project.json").ToList();
-    projectFiles.ForEach(projectFile => 
+
+    var projectGlobs = new List<string>() { SRC_PROJECT_GLOB };
+    var projectFiles = Files.Include(SRC_PROJECT_GLOB).ToList();
```
Comment (pranavkm) [inline build/shade/_k-standard-goals.shade]: `srcProjects`?

File: build/shade/_k-standard-goals.shade [hunk 2]
```diff
@@ -107,30 +109,42 @@ default PACKAGELIST_JSON_FILENAME = 'NuGetPackageVerifier.json'
         GitCommand("rev-parse HEAD >> " + commitHashFile);
         commitHash = File.ReadAllLines(commitHashFile)[0];
     }
-  
-    var projectFiles = Files.Include("src/*/project.json").ToList();
-    projectFiles.ForEach(projectFile => 
+
+    var projectGlobs = new List<string>() { SRC_PROJECT_GLOB };
```
Comment (pranavkm) [inline build/shade/_k-standard-goals.shade]: Would this work if a repo doesn't have `src`? For e.g. ServerTests

File: build/shade/_k-standard-goals.shade [hunk 3]
```diff
@@ -139,16 +153,6 @@ default PACKAGELIST_JSON_FILENAME = 'NuGetPackageVerifier.json'
     }
   }
 
-#build-test target='compile' if='Directory.Exists("test") && !BuildSrcOnly'
```
Comment (pranavkm) [inline build/shade/_k-standard-goals.shade:142]: Is `BuildSrcOnly` still relevant? If not, we could remove it.

**Comment by pranavkm** [comment]
:shipit: when @victorhurdugaci is happy.

File: build/shade/_k-standard-goals.shade (hunk 3 above)
Comment (pakrym) [inline build/shade/_k-standard-goals.shade:142]: I have no idea, where we used it?

File: build/shade/_k-standard-goals.shade [hunk 4]
```diff
@@ -16,6 +16,9 @@ default BUILD_DIR='${Path.Combine(TARGET_DIR, "build")}'
 default TEST_DIR='${Path.Combine(TARGET_DIR, "test")}'
 default Configuration='${E("Configuration")}'
 default PACKAGELIST_JSON_FILENAME = 'NuGetPackageVerifier.json'
+default SRC_PROJECT_GLOB = "src/*/project.json"
+default TEST_PROJECT_GLOB = "test/*/project.json"
```
Comment (victorhurdugaci) [inline build/shade/_k-standard-goals.shade]: I think some repos have multiple folders nested folders under `test`. Some have something like `test\testapps\app1\project.json`

**Comment by victorhurdugaci** [comment]
Looks good. Make sure it doesn't break DataProtection because that one has a custom makefile.

:shipit:

**Comment by pakrym** [comment]
I don't see sane ways of doing this and not breaking `DataProtections`, I would need to fix `DataProtection` by copying part of compile code unfortunately.

**Comment by victorhurdugaci** [comment]
I don't see anything wrong with doing that :)

**Pull request merged**